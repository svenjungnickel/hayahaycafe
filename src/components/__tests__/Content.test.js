import React from 'react';
import renderer from 'react-test-renderer';
import Content, { HTMLContent } from '../Content';

describe('Content', () => {
    it('renders content', () => {
        const content = 'Content';

        const component = <Content content={content} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders classname', () => {
        const className = 'className';

        const component = <Content className={className} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders content and classname', () => {
        const content = 'Content';
        const className = 'className';

        const component = <Content content={content} className={className} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders with invalid content throws prop type error', () => {
        const renderContent = () => {
            const content = {};

            const component = <Content content={content} />;
            renderer.create(component);
        };

        expect(renderContent).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid className throws prop type error', () => {
        const renderContent = () => {
            const content = 'Content';
            const className = 123;

            const component = <Content content={content} className={className} />;
            renderer.create(component);
        };

        expect(renderContent).toThrowError('Warning: Failed prop type');
    });
});

describe('HTMLContent', () => {
    it('renders content', () => {
        const content = '<p>HTMLContent</p>';

        const component = <HTMLContent content={content} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders classname', () => {
        const className = 'className';

        const component = <HTMLContent className={className} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders content and classname', () => {
        const content = '<p>HTMLContent</p>';
        const className = 'className';

        const component = <HTMLContent content={content} className={className} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders with invalid content throws prop type error', () => {
        const renderHTMLContent = () => {
            const content = {};

            const component = <HTMLContent content={content} />;
            renderer.create(component);
        };

        expect(renderHTMLContent).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid className throws prop type error', () => {
        const renderHTMLContent = () => {
            const content = '<p>HTMLContent</p>';
            const className = 123;

            const component = <HTMLContent content={content} className={className} />;
            renderer.create(component);
        };

        expect(renderHTMLContent).toThrowError('Warning: Failed prop type');
    });
});
