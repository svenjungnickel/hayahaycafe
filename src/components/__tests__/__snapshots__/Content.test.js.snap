// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders classname 1`] = `
<div
  className="className"
/>
`;

exports[`Content renders content 1`] = `
<div>
  Content
</div>
`;

exports[`Content renders content and classname 1`] = `
<div
  className="className"
>
  Content
</div>
`;

exports[`HTMLContent renders classname 1`] = `
<div
  className="className"
  dangerouslySetInnerHTML={
    Object {
      "__html": undefined,
    }
  }
/>
`;

exports[`HTMLContent renders content 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>HTMLContent</p>",
    }
  }
/>
`;

exports[`HTMLContent renders content and classname 1`] = `
<div
  className="className"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>HTMLContent</p>",
    }
  }
/>
`;
